### PR TITLE
Move example configs in their own directories

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: component/component.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ type Host interface {
 	// Typically is used to find exporters by type or by full config name. Both cases
 	// can be done by iterating the returned map. There are typically very few exporters
 	// so there there is no performance implications due to iteration.
-	// This returns a map by DataType of maps by exporter configs to the exporter instance.
```
Comment (james-bebbington) [inline component/component.go:79]: Is this change intended (+ a few other similar ones in this PR?)

**Comment by tigrannajaryan** [comment]
@bogdandrutu please resolve conflicts.

File: component/component.go (hunk 1 above)
Comment (bogdandrutu) [reply component/component.go]: done. goland decided to do these changes because of a rename.

**Comment by bogdandrutu** [comment]
@tigrannajaryan @james-bebbington ptal

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,4 @@
 bin/
-local/
```
Comment (owais) [inline .gitignore:2]: Can we keep this? I have almost 50 config files locally to test different scenarios and I'd have to move them out if we remove this. I don't know if anyone else uses this but it is quite harmless in any case. 🙏

**Review by james-bebbington — APPROVED** [review]
LGTM. See @owais's comment before merging though.

**Review by tigrannajaryan — COMMENTED** [review]
@bogdandrutu moving files is LGTM, but see @owais's comment.

File: .gitignore (hunk 2 above)
Comment (bogdandrutu) [reply .gitignore:2]: Not really, unless we change the location for that file

Comment (bogdandrutu) [reply .gitignore:2]: Is either you change your local files or we change where the config is in examples

Comment (owais) [reply .gitignore:2]: Can we rename the ignore rule to `/local` so it only ignores the local dir at project root? If not, can we rename it to anything else? Let's say `local-etc` or something. I don't have a preferences for either as long as there is one gitignored directory in the project we can place helper config/scripts in without checking them in.

Comment (tigrannajaryan) [reply .gitignore:2]: @bogdandrutu thoughts? ^^^ Sounds useful to allow a place for ignored local config files.

Comment (bogdandrutu) [reply .gitignore:2]: That can be achieve simply by having that in the local git config. I am not sure any other project uses that.

**Comment by tigrannajaryan** [comment]
Are we good to merge this or this needs further discussion?

There is a conflict as well, please fix.

**Comment by tigrannajaryan** [comment]
@bogdandrutu @owais this is pending your resolution on the one remaining open comment.

**Comment by owais** [comment]
Not adding `./local` to .gitignore is an inconvenience and I don't see the harm in keeping it but I'm totally fine with removing it as well. I'll just move the file to some other location. So no blocker from my side.

**Comment by tigrannajaryan** [comment]
@bogdandrutu please resolve the conflict.

**Comment by tmc** [comment]
This broke the current content at https://opentelemetry.io/docs/collector/about/ -> https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector/master/examples/k8s.yaml
